src/components/elementView/elementTable.tsx

```diff
@@ -451,12 +457,14 @@ class ElementTable extends React.Component<
 
   updateFilterOrder = (items: Array<IFilter>) => {
     this.setState({
-      filters: items
+      filters: items,
+      curr_page: 1
     });
     this.updateFilterData(items);
   };
-  getFieldNames = () => {
+  setFieldNames = () => {
     let fields: Array<string> = [];
+
     if (this.state.items && this.state.items.length > 0) {
       Object.keys(this.state.items[0]).forEach((col: string) => {
         if (col === "model") {
```

Comment:
Is this the function you were referring to that dynamically renders column names? And if so, why do you only push if col === model?

Reply:
oh, I think the rest of the method was just hidden,I have a special case for model because i need to break it into model__vendor and model__model_number fields. i account for the other columns here :

 ```       fields.push("model__vendor");
          fields.push("model__model_number");
        } else if (col !== "id") {
          fields.push(col);
        }
      });```

Reply:
Also, no this was not the function I was referring to, although this one probably also works for detailview. I added one in utils.tsx that actually calls the backend endpoint that returns the fields. This one gets the fields from the current data. Either one would work.